model/plotWidget.cc

```diff
@@ -488,21 +489,39 @@ namespace minsky
             }
           
           // higher rank y objects treated as multiple y vectors to plot
-          for (size_t j=0 /*d[0]*/; j<std::min(size_t(10)*d[0], yv.numElements()); j+=d[0])
-            {
-              setPen(extraPen, x, yv.begin()+j, d[0]);
-              if (pen>=numLines)
-                assignSide(extraPen,Side::right);
-             string label;
-              size_t stride=d[0];
-              for (size_t i=1; i<yv.xVector.size(); ++i)
-                {
-                  label+=str(yv.xVector[i][(j/stride)%d[i]])+" ";
-                  stride*=d[i];
-                }
-              labelPen(extraPen,label);
-              extraPen++;
-            }
+          // For feature 47
+          if (yv.index.empty())
+            for (size_t j=0 /*d[0]*/; j<std::min(size_t(10)*d[0], yv.numDenseElements()); j+=d[0])
+              {
+                setPen(extraPen, x, yv.begin()+j, d[0]);
+                if (pen>=numLines)
+                  assignSide(extraPen,Side::right);
+               string label;
+                size_t stride=d[0];
+                for (size_t i=1; i<yv.xVector.size(); ++i)
+                  {
+                    label+=str(yv.xVector[i][(j/stride)%d[i]])+" ";
+                    stride*=d[i];
+                  }
+                labelPen(extraPen,label);
+                extraPen++;
+              }
+           else
+            for (size_t j=0 /*d[0]*/; j<std::min(size_t(10)*d[0], yv.numSparseElements); j+=d[0])
```

Comment:
In this branch, we need to unroll the index vector entry to pick up the pen label.

But we should aim to make this consideration as invisible as possible in this client code, so think about how to write a method that returns the hypercube index for a given offset into the actual data.

Reply:
The index vector at present contains the serialized or flattened hypercube indices corresponding the actual data, so it is immediately available, but I'm not sure that is correct. As alluded to above, this is where my greatest confusion about the sparse data implementation lies.